internal/winapi/cimfs.go

```diff
@@ -0,0 +1,45 @@
+package winapi
+
+import (
+	"syscall"
+	"unsafe"
+
+	"github.com/Microsoft/go-winio/pkg/guid"
+)
+
+type g = guid.GUID
+type FsHandle uintptr
+type StreamHandle uintptr
+
+type CimFsFileMetadata struct {
+	Attributes uint32
+	FileSize   int64
+
+	CreationTime   syscall.Filetime
+	LastWriteTime  syscall.Filetime
+	ChangeTime     syscall.Filetime
+	LastAccessTime syscall.Filetime
+
+	SecurityDescriptorBuffer unsafe.Pointer
+	SecurityDescriptorSize   uint32
+
+	ReparseDataBuffer unsafe.Pointer
+	ReparseDataSize   uint32
+
+	ExtendedAttributes unsafe.Pointer
+	EACount            uint32
+}
+
+//sys CimMountImage(imagePath string, fsName string, flags uint32, volumeID *g) (hr error) = cimfs.CimMountImage?
+//sys CimDismountImage(volumeID *g) (hr error) = cimfs.CimDismountImage?
+
+//sys CimCreateImage(imagePath string, oldFSName *uint16, newFSName *uint16, cimFSHandle *FsHandle) (hr error) = cimfs.CimCreateImage?
```

Comment:
how come we use *uint16 for some of the PWSTR fields but we don't for imagePath here?

Reply:
That's because this `CimCreateImage` API differentiates between an empty value vs `nil` value for these fields. But if you set these fields as strings there is no way to pass a nil for that.